mm2src/coins/eth/v2_activation.rs

```diff
@@ -916,3 +921,38 @@ fn compress_public_key(uncompressed: H520) -> MmResult<H264, EthActivationV2Erro
     let compressed = public_key.serialize();
     Ok(H264::from(compressed))
 }
+
+#[cfg(not(target_arch = "wasm32"))]
+async fn run_db_migration_for_new_eth_pubkey(ctx: &MmArc, db_id: String) -> MmResult<(), EthActivationV2Error> {
+    info!("Public key hash: {db_id:?}");
+
+    let mut db_migration_sender = ctx
+        .db_migration_watcher
+        .as_option()
+        .expect("Db migration watcher isn't intialized yet!")
+        .get_sender();
+    db_migration_sender
+        .send(db_id)
+        .await
+        .map_to_mm(|err| EthActivationV2Error::InternalError(err.to_string()))?;
+
+    Ok(())
+}
+
+pub(super) async fn eth_shared_db_id(coin: &EthCoin, ctx: &MmArc) -> Option<String> {
+    // Use the hd_wallet_rmd160 as the db_id in HD mode since it's unique to a device and not tied to a single address
+    coin.derivation_method()
+        .hd_wallet()
+        .map(|_| ctx.default_shared_db_id().to_string())
+}
+
+pub(super) async fn eth_account_db_id(coin: &EthCoin) -> Option<String> {
+    if let Some(hd_wallet) = coin.derivation_method().hd_wallet() {
+        return hd_wallet
+            .get_enabled_address()
+            .await
+            .map(|addr| hex::encode(dhash160(addr.pubkey().as_bytes())));
+    }
+
+    None
```

Comment:
Wouldn't it be better to match here instead of using `if let` in case we add a new derivation method. I guess for WalletConnect (external wallets) we will follow a similar approach Tendermint and remove derivation method and make it `activation_policy` enum instead that contains derivation method as a variant. So no need to match if we are gonna change this, since we will remember to handle external wallets db_id in such case.

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/blob/d57e39404ed491264c05dfec1f9fc7a55edc4356/mm2src/coins/eth/v2_activation.rs#L1015-L1021
Using a wildcard is the same as using if let, we should explicitly match all cases.

Reply:
Ikr .. less edit when matching other case :xD